bluetooth/update/README.md

```diff
@@ -2,145 +2,173 @@
+FROM lines
+GROUP BY analysis_id, street, direction, from_street, to_street) a)
 ```
 
-Matches the centreline road segment to the drawn lines. Note the buffer is really large (1000 units) to account for segments that run accross multiple roads.
-
-After proper QC (see next section) is applied, it should something like this.
-
-![bt_segments](img/bt_segments.PNG)
-
-## Cleanup, Warnings, and Alternative Methods
+![bt_new_segments](img/new_segments.JPG)
 
-This method works really well for straight roads/segments. However, this is especially challenging for segments that are perpendicular, involves curves, or weird geometries/routes. 
+## Things to note 
+A number of centreline need to be excluded during routing, for example: Geostatistical lines and planning boundaries. Those can be filtered using the following where clause: 
+```sql
+WHERE fcode_desc IN ('Collector','Collector Ramp','Expressway','Expressway Ramp',
+'Local','Major Arterial','Major Arterial Ramp','Minor Arterial',
+'Minor Arterial Ramp','Pending')
+``` 
 
-A lot of mannual cleanup was required to delete extraneous roads, or add roads to the route, mainly through the `UPDATE` and `DELETE` clauses and using the `geo_ids` from the centreline table.
+## Validating Output
+Validate the length of the segments with length `ST_length(geom)` and direction using `gis.direction_from_line(geom)` functions. If the detectors are located very close to the centerline intersections, it is not necessary to do the centreline cutting. If any bluetooth detectors are not located at the start or end point of a centreline, we will need to cut the centreline using `ST_linesubstring()` as explained in [here.](https://github.com/CityofToronto/bdit_data-sources/issues/234).  
 
-An alternative method is drawing routes in QGIS connecting the readers on the centreline layer. This may be faster for low number of segments (ex <500).
-
-## Making Lines
+Steps to cut centreline using `ST_linesubstring()`:
+1) Find the closest point of the detector on the centreline with [ST_closespoint()](https://postgis.net/docs/ST_ClosestPoint.html).   
+```sql
+ST_closestpoint(detector_geom, centreline_geom)
+```
+2) Return the location of the point relative to the centreline using [`ST_linelocatepoint()`](https://postgis.net/docs/ST_LineLocatePoint.html)
+```sql
+ST_linelocatepoint(geom, closest_point_geom)
+```
+3) Cut the line using [`ST_linesubstring()`](https://postgis.net/docs/ST_LineSubstring.html) 
+```sql
+ST_linesubstring(geom, 0, st_linelocatepoint)
+```
 
-At this point, the layer has multiple entries containing centreline segments for each analysis_id. Instead, we need a single line/entry for each analysis_id. This query concatenates the individual centreline segments into one line. The main function doing this work is [`ST_LineMerge()`](https://postgis.net/docs/ST_LineMerge.html)
+The new routes table is now ready to append to the existing routes table.  
```

Comment:
The `bluetooth.segments` table has the columns:     

- segment_name ,
- analysis_id,
- street ,
- direction,
- start_cross ,
- end_cross ,
- length ,
- bluetooth ,
- wifi ,
- geom ,
- duplicate ,
- reversed ,
- end_street

So its missing: 
- segment_name , bluetooth, widi, duplicate, revered, end_street